crates/bevy_ui/src/flex/convert.rs

```diff
@@ -224,3 +284,73 @@ impl From<FlexWrap> for taffy::style::FlexWrap {
         }
     }
 }
+
+impl From<GridAutoFlow> for taffy::style::GridAutoFlow {
+    fn from(value: GridAutoFlow) -> Self {
+        match value {
+            GridAutoFlow::Row => taffy::style::GridAutoFlow::Row,
+            GridAutoFlow::RowDense => taffy::style::GridAutoFlow::RowDense,
+            GridAutoFlow::Column => taffy::style::GridAutoFlow::Column,
+            GridAutoFlow::ColumnDense => taffy::style::GridAutoFlow::ColumnDense,
+        }
+    }
+}
+
+impl From<GridPlacement> for taffy::geometry::Line<taffy::style::GridPlacement> {
+    fn from(value: GridPlacement) -> Self {
+        let start = match value.start {
+            None => taffy::style::GridPlacement::Auto,
+            Some(start) => style_helpers::line(start as i16),
+        };
+        let span = taffy::style::GridPlacement::Span(value.span);
+        taffy::geometry::Line { start, end: span }
+    }
+}
+
+impl From<MinTrackSizingFunction> for taffy::style::MinTrackSizingFunction {
+    fn from(value: MinTrackSizingFunction) -> Self {
+        match value {
+            MinTrackSizingFunction::Fixed(val) => {
+                taffy::style::MinTrackSizingFunction::Fixed(val.into())
+            }
+            MinTrackSizingFunction::Auto => taffy::style::MinTrackSizingFunction::Auto,
+            MinTrackSizingFunction::MinContent => taffy::style::MinTrackSizingFunction::MinContent,
+            MinTrackSizingFunction::MaxContent => taffy::style::MinTrackSizingFunction::MaxContent,
+        }
+    }
+}
+
+impl From<MaxTrackSizingFunction> for taffy::style::MaxTrackSizingFunction {
+    fn from(value: MaxTrackSizingFunction) -> Self {
+        match value {
+            MaxTrackSizingFunction::Fixed(val) => {
+                taffy::style::MaxTrackSizingFunction::Fixed(val.into())
```

Comment:
The numeric values here need to be multiplied by the scale factor.

I think my implementation of `Into` for `Val` was a mistake, it shouldn't be possible to do the conversion without the scale_factor, it's too easy to miss otherwise.

Reply:
Something like:
```rust
impl Val {
    fn scale_by(self, scale_factor: f64) -> Self {
        match self {
            Val::Auto => Val::Auto,
            Val::Percent(value) => Val::Percent(value),
            Val::Px(value) => Val::Px((scale_factor * value as f64) as f32),
            Val::Undefined => Val::Undefined,
        }
    }

    fn to_dimension(self, scale_factor: f64) -> taffy::style::Dimension {
        match self.scale_by(scale_factor) {
            Val::Auto | Val::Undefined => taffy::style::Dimension::Auto,
            Val::Percent(value) => taffy::style::Dimension::Percent(value / 100.0),
            Val::Px(value) => taffy::style::Dimension::Points(value),
        }
    }
    
    fn to_length_percentage(self, scale_factor: f64) -> taffy::style::LengthPercentage {
        match self.scale_by(scale_factor) {
            Val::Auto | Val::Undefined => taffy::style::LengthPercentage::Points(0.0),
            Val::Percent(value) => taffy::style::LengthPercentage::Percent(value / 100.0),
            Val::Px(value) => taffy::style::LengthPercentage::Points(value),
        }
    }

    fn to_length_percentage_auto(self, scale_factor: f64) -> taffy::style::LengthPercentageAuto {
        match self.scale_by(scale_factor) {
            Val::Auto => taffy::style::LengthPercentageAuto::Auto,
            Val::Percent(value) => taffy::style::LengthPercentageAuto::Percent(value / 100.0),
            Val::Px(value) => taffy::style::LengthPercentageAuto::Points(value),
            Val::Undefined => taffy::style::LengthPercentageAuto::Points(0.),
        }
    }

    fn to_inset(self, scale_factor: f64) -> taffy::style::LengthPercentageAuto {
        match self.scale_by(scale_factor) {
            Val::Auto | Val::Undefined => taffy::style::LengthPercentageAuto::Auto,
            Val::Percent(value) => taffy::style::LengthPercentageAuto::Percent(value / 100.0),
            Val::Px(value) => taffy::style::LengthPercentageAuto::Points(value),
        }
    }
}
```